website/docs/cloud-docs/integrations/kubernetes/k8s-ops-v2-migration.mdx

```diff
@@ -0,0 +1,104 @@
+$ git clone https://github.com/hashicorp/terraform-cloud-operator.git
+```
+
+Change to the migration directory.
+
+```shell-session
+$ cd terraform-cloud-operator/docs/migration/crds
+```
+
+View the changes that **patch A** will apply to the workspace CRD.
+
+```shell-session
+$ kubectl diff -f workspaces_patch_a.yaml
+```
+
+Patch the workspace CRD with **patch A**. This patch adds `app.terraform.io/v1alpha2` support, but excludes `.status.runStatus` because it has a different format in `app.terraform.io/v1alpha1` and causes JSON unmarshalling issues.
+
+```shell-session
+$ kubectl patch crd workspaces.app.terraform.io --patch-file workspaces_patch_a.yaml
+```
+
+Follow the [installation guide](https://github.com/hashicorp/terraform-cloud-operator/blob/main/docs/installation.md) to install v2 of the operator.
+
+Next, create a Kubernetes secret to store the TFC API token following the [Usage Guide](https://github.com/hashicorp/terraform-cloud-operator/blob/main/docs/usage.md#prerequisites). The API token can be copied from the Kubernetes secret that you created for v1 of the operator. By default, this is named `terraformrc`. Use the `kubectl get secret` command to get the API token.
+
+```shell-session
+$ kubectl --namespace ${RELEASE_NAMESPACE} get secret terraformrc -o json | jq '.data.credentials' | tr -d '"' | base64 -d
+```
+
+If v2 of the operator pod fails to start, ensure that each `workspace` object has the `spec.token` value set. To do this, patch each `workspace` object using the `kubectl patch` command.
+
+```shell-session
+$ kubectl patch workspace <WORKSPACE_NAME> --type=merge --patch '{"spec": {"terraformVersion": "1.4.6", "token": {"secretKeyRef": {"key": "token", "name": "tfc-operator"}}}}'
+```
+
+Validate
+
+```shell-session
+$ kubectl logs -f <POD_NAME>
+```
+
+View the changes that **patch B** will apply to the workspace CRD:
+
+```shell-session
+$ kubectl diff -f workspaces_patch_b.yaml
+```
+
+Patch workspace CRD with **patch B**. This patch adds `.status.runStatus` support that was excluded in Patch A.
+
+```shell-session
+$ kubectl patch crd workspaces.app.terraform.io --patch-file workspaces_patch_b.yaml
+```
+
+The v2 operator will fail to proceed if a custom resource has the v1 finalizer `finalizer.workspace.app.terraform.io`. If you encounter an error, check the logs for more information.
+
+```shell-session
+$ kubectl logs -f <POD_NAME>
+```
+
+Specifically, look for an error message such as the following.
+
+```
+ERROR	Migration	{"workspace": "default/<WORKSPACE_NAME>", "msg": "spec contains old finalizer finalizer.workspace.app.terraform.io"}
+```
+The `finalizer` exists to provide greater control over the migration process. Verify the custom resource, and when you’re ready to migrate it, use the `kubectl patch` command to update the `finalizer` value.
+
+```shell-session
+$ kubectl patch workspace migration --type=merge --patch '{"metadata": {"finalizers": ["workspace.app.terraform.io/finalizer"]}}'
+```
```

Comment:
Are we done? Do we need to re-run the `kubectl patch` command if it failed the first time bc of the v1 finalizer?

Regardless, I think we should say something to let the user know they are done, and maybe validate that it worked?